jupyterhub/apihandlers/users.py

```diff
@@ -162,17 +162,20 @@ def get(self):
             )
         elif state_filter:
             raise web.HTTPError(400, f"Unrecognized state filter: {state_filter!r}")
+        else:
+            # need a join on Spawner for contains_eager below
+            query = query.outerjoin(orm.Spawner, orm.User._orm_spawners)
 
         # apply eager load options
         query = query.options(
             selectinload(orm.User.roles),
             selectinload(orm.User.groups),
-            joinedload(orm.User._orm_spawners).joinedload(orm.Spawner.user),
+            contains_eager(orm.User._orm_spawners).selectinload(orm.Spawner.user),
             # raiseload here helps us make sure we've loaded everything in one query
             # but since we share a single db session, we can't do this for real
             # but it's useful in testing
             # raiseload("*"),
-        )
+        ).populate_existing()
```

Comment:
Whenever using `contains_eager` with filtering, we need to `populate_existing`: 
- https://github.com/sqlalchemy/sqlalchemy/issues/5740#issuecomment-737240021
- https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#using-contains-eager-to-load-a-custom-filtered-collection-result